hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieAvroDataBlock.java

```diff
@@ -224,6 +249,159 @@ public IndexedRecord next() {
     }
   }
 
+  /**
+   * StreamingRecordIterator is an iterator for reading records from a Hoodie log block in a streaming manner.
+   * It reads data from a given input stream, decodes Avro records, and supports schema promotion.
+   *
+   * This iterator ensures that the buffer has enough data for each record and handles buffer management,
+   * including compaction and resizing when necessary.
+   */
```

Comment:
```java
/**
   * {@code StreamingRecordIterator} is an iterator for reading records from a Hoodie log block in streaming manner.
   * It decodes the given input stream into Avro records with optional schema promotion.
   *
   * <p>This iterator ensures that the buffer has enough data for each record and handles buffer setup,
   * including compaction and resizing when necessary.
   */
```

Reply:
> ```java
> /**
>    * {@code StreamingRecordIterator} is an iterator for reading records from a Hoodie log block in streaming manner.
>    * It decodes the given input stream into Avro records with optional schema promotion.
>    *
>    * <p>This iterator ensures that the buffer has enough data for each record and handles buffer setup,
>    * including compaction and resizing when necessary.
>    */
> ```

Wow, thanks a lot for helping me add the comments! I'll pay more attention to these requirements next time, haha